src/jit/compiler.cpp

```diff
@@ -2293,13 +2302,24 @@ void Compiler::compSetProcessor()
         if (JitConfig.EnableAVX() != 0)
         {
             opts.compCanUseAVX = true;
-            if (!compIsForInlining())
-            {
-                codeGen->getEmitter()->SetUseAVX(true);
-            }
         }
     }
 #endif // FEATURE_AVX_SUPPORT
+
+    if (!compIsForInlining())
+    {
+#ifdef FEATURE_AVX_SUPPORT
+        if (opts.compCanUseAVX)
```

Comment:
This needs to be examined to know the value addition this #ifdef has.

It is ON by default on RyuJIT xarch.
It is not ON on legacy backend (particularly emitter code paths).

---
In reply to: [89870838](https://github.com/dotnet/coreclr/pull/8291#discussion_r89870838) [](ancestors = 89870838)